steam/client/__init__.py

```diff
@@ -161,6 +173,7 @@ def _handle_logon(self, msg):
         if result == EResult.OK:
             self._reconnect_backoff_c = 0
             self.logged_on = True
+            self.cell_id = msg.body.cell_id
```

Comment:
I'm happy to be corrected, but there is no point of holding the value of cell_id as it not needed for anything. By default 0 is used, which lets the CM auto detected from source IP.

Reply:
`cell_id` is a required parameter in the content server discovery process. Using `0` as `cell_id` returns random ones and therefore the actual logged in cell_id is needed to get the lowest possible latency.

Reply:
`cell_id = 0` lets steam choose the servers that it thinks are best, usually by geoip or possible load balance.  I don't see no point in setting it, but it could be exposed, so that it can be set before login.

Reply:
Not sure if its a bug or intended behaviour on Valves side but if you pass `cell_id = 0` on that particular endpoint it will return random servers.

Reply:
random as in from any cell id?

Reply:
Exactly

Reply:
That doesn't sound right